VogtObserver/Date.cs

```diff
@@ -7,10 +7,12 @@ namespace VogtObserver
     class Date : IObserver
     {
         private DateTime _dateTime = DateTime.Now;
+        private readonly RandomNumberGenerator _randomNumberGenerator;
 
-        public Date()
+        public Date(RandomNumberGenerator randomNumberGenerator)
         {
-            
+            _randomNumberGenerator = randomNumberGenerator;
+            _randomNumberGenerator.Add(this);
```

Comment:
You usually don't .Add(this). Thats usually a code smell

Reply:
Dont pass in RandomNumberGenerator in constructor. See my comment about not using dependency injection on the observable/observer relationship specifically. Your goal is to abstract the observer from the observable, and vice versa.